src/twisted/internet/endpoints.py

```diff
@@ -888,6 +1016,22 @@ def cancelRemainingPending(result):
         return d
 
 
+    def _nameResolution(self, host, port):
+        """
+        Resolve the hostname string into a tuple containing the host
+        address.
```

Comment:
It would probably be good to make it clear at least in the docstring and perhaps in the method name that this method is just used as a fallback when the reactor provided to the endpoint doesn't implement the new interface.